c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -688,6 +834,69 @@ public:
     });
   }
 
+  kj::Promise<void> connect(ConnectContext context) override {
+    auto params = context.getParams();
+    auto host = params.getHost();
+    auto headers = factory.headersToKj(params.getHeaders());
+    auto pipe = kj::newTwoWayPipe();
+
+    class EofDetector final: public kj::AsyncOutputStream {
+    public:
+      EofDetector(kj::Own<kj::AsyncIoStream> inner) : inner(kj::mv(inner)) {}
+      ~EofDetector() {
+        inner->shutdownWrite();
+      }
+
+      kj::Maybe<kj::Promise<uint64_t>> tryPumpFrom(
+          kj::AsyncInputStream& input, uint64_t amount = kj::maxValue) override {
+        return inner->tryPumpFrom(input, amount);
+      }
+
+      kj::Promise<void> write(const void* buffer, size_t size) override {
+        return inner->write(buffer, size);
+      }
+
+      kj::Promise<void> write(kj::ArrayPtr<const kj::ArrayPtr<const byte>> pieces) override {
+        return inner->write(pieces);
+      }
+
+      kj::Promise<void> whenWriteDisconnected() override {
+        return inner->whenWriteDisconnected();
+      }
+    private:
+      kj::Own<kj::AsyncIoStream> inner;
+    };
+
+    kj::Own<kj::AsyncOutputStream> stream = factory.streamFactory.capnpToKj(
+        context.getParams().getDown());
+    auto promises = kj::heapArrayBuilder<kj::Promise<void>>(2);
+    // We write to the `down` pipe.
+    promises.add(pipe.ends[1]->pumpTo(*stream).then([context](uint64_t) mutable {
+      auto endReq = context.getParams().getDown().endRequest();
```

Comment:
Feels better to capture only what we need here (and capture a strong reference rather than a weak one -- capturing `context` is effectively capturing an unowned pointer).

```suggestion
    promises.add(pipe.ends[1]->pumpTo(*stream).then([down = params.getDown()](uint64_t) mutable {
      auto endReq = down.endRequest();
```

Reply:
Wait, I think this is redundant anyway. The `stream` will automatically call `end()` when you drop it. So I think this whole `.then()` should just be deleted -- the `.attach(kj::mv(stream))` will do the job.